docs/notes/2.23.x.md

```diff
@@ -57,6 +57,10 @@ silently ignored.
 
 Fixed a bug where the internal Docker BuildKit parser would return `<unknown> image_id` if the BuildKit output used step durations.
 
+#### CC
+
+Migrated the `cc` backend to use the new `call-by-name` syntax.
```

Comment:
Does this need to be in the release notes? It is of no consequence to the end user.

Reply:
It's just that I'm not sure what an end user is supposed to do with this information. It adds cognitive load - "what is this new thing? Do I need to care? Does it affect me?" So if we do keep this we should elaborate that they do not need to care about this unless they notice a new and inexplicable bug in this backend.

Reply:
The phrasing I used was "The internal code for exporting JVM tools was refactored."

Reply:
>  It adds cognitive load - "what is this new thing? Do I need to care? Does it affect me?"

I think a casual user who is not using the backend won't care. I think for casual users who are the kind to read release notes, it's an incremental burden.

BUT, as an out-of-repo plugin developer, this is the kinda thing I'd want to know about. A backend-wide change that might (hopefully not) affect the stability of my external plugins. A perfect example is Svelte upgrading all of their internals for Svelte 5. In theory, it shouldn't affect me - but in practice, a bunch of my stuff will be broken in the short term - and it's good to call that kinda stuff out.

I can use something closer to what @lilatomic suggested, because the more I think about it - the more important it should be called out "somewhere" and the release notes are where it makes the most sense. A release manager can feel free to cull at release time, but I'd rather leave that decision up to the release manager. 

"Migrated the `cc` backend to use the new `call-by-name` syntax." -> "The internal code for calling `@rule`s was migrated to the new call-by-name syntax. `cc` backend users should be unaffected"?

Reply:
That makes it sounds like `cc` was not migrated... And also, we will likely migrate all backends by the next release, so not sure we need to call out `cc` specifically just because it's the first?

Maybe just a special note:

```
### New call-by-name syntax for @rules

Pants has a new mechanism for @rule invocation in backends. In this release all built-in backends were migrated to use this new mechanism. There should not be any user-visible effect, but please be on the lookout for any unusual bugs or error messages.
```

Reply:
> we will likely migrate all backends by the next release

Ambitious 😆 

I've run into a bunch of cyclic deps, and other weirdness - I'll use your line, and call it the `cc` backend, and that can be updated as other PRs filter in.